src/components/avatar/Avatar.vue

```diff
@@ -0,0 +1,138 @@
+        username: String,
+        alt: {
+            type: String,
+            default: 'This is avatar for the user'
+        },
+        rounded: {
+            type: Boolean,
+            default: config.defaultAvatarRounded
+        },
+        size: {
+            type: String,
+            default: config.defaultAvatarSize
+        },
+        type: String
+    },
+    data() {
+        return {
+            defaultAvatarSize: config.defaultAvatarSize,
+            defaultVariant: config.defaultAvatarVariant,
+            defaultAvatarRounded: config.defaultAvatarRounded,
+            dynamicClass: this.username + this.size,
+            allowedTextSizes: {
+                // Text sizes are used to be compatible to appropriate avatar size
+                // specified by the user
+                size16: '0.5rem',
+                size24: '0.625rem',
+                size32: '0.9375rem',
+                size48: '1.20rem',
+                size64: '1.525rem',
+                size96: '2.1rem',
+                size128: '2.8125rem'
+            },
+            bulmaKnownSizes: ['16x16', '24x24', '32x32', '48x48', '64x64', '96x96', '128x128'],
+            bulmaKnownColors: [
+                'has-background-black',
+                'has-background-dark',
+                'has-background-primary',
+                'has-background-link',
+                'has-background-info',
+                'has-background-success',
+                'has-background-warning',
+                'has-background-danger'
+            ]
+        }
+    },
+    computed: {
+        imgClasses() {
+            // check if the size known by bulma
+            const bulmaKnownSize =
+              this.bulmaKnownSizes.includes(this.size) ? this.size : this.defaultAvatarSize
+            //
+            if (this.rounded && this.size) {
+                this.calculateTextSize()
+                return `image rounded is-${bulmaKnownSize}`
+            } else if (!this.rounded) {
+                this.calculateTextSize()
+                return `image is-${bulmaKnownSize}`
+            }
+        },
+        generateBgColorForAvatar() {
+            if (!this.src && this.username && this.type) {
+                // check if the size known by bulma
+                const bulmaKnownColor =
+                  this.bulmaKnownColors.includes(this.type) ? this.type : this.defaultVariant
+                return bulmaKnownColor
+            } else {
+                // Generate random color if the type wasn't provided
+                const type =
+                  this.bulmaKnownColors[Math.ceil(Math.random() * 10)] || this.defaultVariant
+                return type
+            }
+        },
+        setText() {
+            if (!this.src && this.username) {
+                return this.username.substr(0, 2).toUpperCase()
+            } else {
+                return ``
+            }
+        }
+    },
+    methods: {
+        calculateTextSize: function () {
```

Comment:
I didn't try the component but why the size depends on text ?

Reply:
If the avatar set for example to 48x48px  then the text size would be automatically set 1.20rem which is 19px, that means the size of the text is relevant to avatar size (see above example in screenshot).

Reply:
if anything needs to be changed I can do it ASAP